sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/Security/Ucfg/UcfgBuilder_Instructions.cs

```diff
@@ -787,6 +787,58 @@ public void Foo(Class1 other)
 
         }
     }
+}";
+            UcfgVerifier.VerifyInstructions(code, "Foo");
+        }
+        
+        [TestMethod]
+        public void ArrayCreation_WithNew()
+        {
+            const string code = @"
+namespace Namespace
+{
+    public class Class1
+    {
+        public void Foo()
+        {
+            // Primitive type
+            var x0 = new string[42];    // %0 := new string[]
+                                        // x0 := __id [ %0 ]
+
+            // Array of objects
+            var x1 = new Class1[0];     // %1 := new Namespace.Class1[]
+                                        // x1 := __id [ %1 ]
+
+            // Multi-rank arrays
+            var x2 = new int[10,2];     // %2 := new int[*,*]
+                                        // x2 := __id [ %2 ]
+
+            // Arrau with initializer (initializer is ignored)

```

Comment:
Oops, typo in "arrau".
The initializer will be handled in a separate ticket (161).